src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +432,95 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
```

Comment:
Hmm,I saw the await Task.Delay below, and thought you were still in a timer-based polling loop.  Maybe not.  Let me re-read.